include/bluetooth/bluetooth.h

```diff
@@ -698,6 +698,41 @@ static inline int bt_addr_le_to_str(const bt_addr_le_t *addr, char *str,
 			addr->a.val[2], addr->a.val[1], addr->a.val[0], type);
 }
 
+/**
+ * @brief      Convert Bluetooth data from ':' separated data string
+ *
+ * @param[in]  str        Bluetooth data in ':' separated string format
+ * @param      data       Address of data buffer with enough room to store the
+ * formatted data.
+ * @param[in]  data_size  Length of the data buffer.
+ *
+ * @return     { description_of_the_return_value }
+ */
+int bt_data_from_str(const char *str, u8_t *data, u8_t data_size);
```

Comment:
We already have a data type called bt_data, and it's used to refer to the encoding format of BR/EDR EIR and LE advertising data (see `struct bt_data` and `bt_le_adv_*()` parameters in bluetooth.h), so I think it's rather confusing to introduce an API prefixed with bt_data which is for something completely different.

Also, do we really want to use the colon delimited hex representation for anything else than addresses?

Reply:
I'd also like to add that the reason such a generic term is used for AD, SD, EIR and OOB data encoding is that neither the core spec nor the core spec supplement give a better name for it (e.g. the CSS refers just to "data types"), which is why bt_data was originally chosen.

Reply:
I was using it for channel map in the shell, but I can remove that, but I can change that, and then remove this function again.